wagtail/wagtailsearch/backends/elasticsearch.py

```diff
@@ -708,19 +712,21 @@ def reset_index(self):
         self.get_rebuilder().reset_index()
 
     def add_type(self, model):
-        self.get_index().add_model(model)
+        self.get_index_for_model(model).add_model(model)
 
     def refresh_index(self):
         self.get_index().refresh()
 
     def add(self, obj):
-        self.get_index().add_item(obj)
+        self.get_index_for_model(type(obj)).add_item(obj)
 
     def add_bulk(self, model, obj_list):
-        self.get_index().add_items(model, obj_list)
+        self.get_index_for_model(model).add_items(model, obj_list)
 
     def delete(self, obj):
-        self.get_index().delete_item(obj)
+        self.get_index_for_model(type(obj)).delete_item(obj)
 
+# Backwards compatibility
+ElasticSearch = ElasticsearchSearchBackend
```

Comment:
I decided against this because it would be tricky (but not impossible) to do, because of the different places it can be called from require different messages.

But if it must be done, it must be done.